### PR TITLE
Add example with vanilla-extract

### LINKED ISSUE #10: Provide examples for practical use.
- [x] [emotion(styled) x cloudflare](https://github.com/aiji42/remix-esbuild-override/tree/main/examples/emotion-cloudflare)
- [x] [emotion(css props) x cloudflare](https://github.com/aiji42/remix-esbuild-override/tree/main/examples/emotion-css-props-cloudflare)
- [x] [chakra-ui x cloudflare](https://github.com/aiji42/remix-esbuild-override/tree/main/examples/chakra-ui-cloudflare)
- [ ] preact x Runtime is anywhere
- [x] ~~supabase-js x cloudflare~~
    - With the next release of remix, changes to mainFields will no longer be necessary. https://github.com/remix-run/remix/pull/1977
- [x] [mantine x cloudflare](https://github.com/aiji42/remix-esbuild-override/tree/main/examples/mantine-cloudflare)
- [ ] stylus x Runtime is anywhere
- [x] [linaria x Runtime is anywhare](https://github.com/aiji42/remix-esbuild-override/tree/main/examples/linaria) https://github.com/remix-run/remix/issues/1619
- [ ] vanilla-extract x Runtime is anywhere https://github.com/seek-oss/vanilla-extract/discussions/178
- [ ] ~~Buffer polyfill (using @magic-sdk/admin v[1.3.4](https://www.npmjs.com/package/@magic-sdk/admin/v/1.3.4))~~

If you have other needs, let me know.

### PR DESCRIPTION
**Pull request by jbovenschen**
Thanks for creating this packages, with this package I was able to make vanilla-extract and remix.run working together with a custom esbuild plugin. So I decided to contribute the code to your examples. 

Let me know if you want me to change something?

### CODE REVIEW

File: examples/vanilla-extract/package.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    "dev": "remix dev",
+    "start": "remix-serve build",
+    "typecheck": "tsc -b",
+    "postinstall": "remix-esbuild-override"
+  },
+  "dependencies": {
+    "@parcel/css": "^1.10.1",
+    "@remix-run/node": "^1.3.4",
+    "@remix-run/react": "^1.3.4",
+    "@remix-run/serve": "^1.3.4",
```
Comment (aiji42) [inline examples/vanilla-extract/package.json:19]: Specify the latest version possible.

File: examples/vanilla-extract/wrangler.toml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+name = "remix-cloudflare-workers"
```
Comment (aiji42) [inline examples/vanilla-extract/wrangler.toml:1]: This file is not necessary since it is running by `@remix-run/serve`.

**Comment by aiji42** [comment]
@jbovenschen Your contribution is greatly appreciated.
I tried to create a `vanilla-extract` example before, but gave up because I am not familiar with it. I am happy to be able to add to the collection of examples.

**Comment by aiji42** [comment]
Add a link to examples in the README of the project root.

**Review by aiji42 — COMMENTED** [review]
Please correct the areas I commented on.

**Comment by jbovenschen** [comment]
@aiji42 Thanks for the review, made the changes you requested.

**Review by aiji42 — APPROVED** [review]
Approved 🎉

**Comment by aiji42** [comment]
I'm merging this.

**Comment by aiji42** [comment]
Thank you for your wonderful contribution.

**Pull request merged**